src/wallet/external_signer_scriptpubkeyman.cpp

```diff
@@ -49,15 +51,26 @@ ExternalSigner ExternalSignerScriptPubKeyMan::GetExternalSigner() {
     return signers[0];
 }
 
-bool ExternalSignerScriptPubKeyMan::DisplayAddress(const CScript scriptPubKey, const ExternalSigner &signer) const
+std::optional<bilingual_str> ExternalSignerScriptPubKeyMan::DisplayAddress(const CTxDestination& dest, const ExternalSigner &signer) const
 {
     // TODO: avoid the need to infer a descriptor from inside a descriptor wallet
+    CScript scriptPubKey = GetScriptForDestination(dest);
     auto provider = GetSolvingProvider(scriptPubKey);
     auto descriptor = InferDescriptor(scriptPubKey, *provider);
 
-    signer.DisplayAddress(descriptor->ToString());
-    // TODO inspect result
-    return true;
+    UniValue result = signer.DisplayAddress(descriptor->ToString());
+
+    const UniValue& error = find_value(result, "error");
+    if (error.isStr()) return strprintf(_("Signer returned error: %s"), error.getValStr());
+
+    const UniValue& ret_address = find_value(result, "address");
+    if (!ret_address.isStr()) return _("Signer did not echo address");
+
+    if (ret_address.getValStr() != EncodeDestination(dest)) {
+        return strprintf(_("Signer echoed unexpected address %s"), ret_address.getValStr());
+    }
```

Comment:
Functional tests of these errors would be nice, if feasible.

Reply:
This particular one should be doable.

Reply:
Added.